toucan_connectors/http_api/pagination_configs.py

```diff
@@ -0,0 +1,128 @@
+        if len(result) < self.limit:
+            return None
+        else:
+            return self.model_copy(update={"offset": self.offset + self.limit})
+
+
+class PageBasedPaginationConfig(PaginationConfig):
+    page_name: str = "page"
+    page: int = 0
+
+    def apply_pagination_to_data_source(self, data_source: HttpAPIDataSource) -> HttpAPIDataSource:
+        page_based_params = {self.page_name: self.page}
+        if data_source.params is None:
+            data_source.params = page_based_params
+        else:
+            data_source.params |= page_based_params
+        return data_source
+
+    def get_next_pagination_config(
+        self, result: Any, pagination_info: Any | None
+    ) -> Optional["PageBasedPaginationConfig"]:
+        if len(result) < 1:
+            return None
+        else:
+            return self.model_copy(update={"page": self.page + 1})
+
+
+class CursorBasedPaginationConfig(PaginationConfig):
+    cursor_name: str = "cursor"
+    cursor: str | None = None
+    cursor_filter: str = FilterSchema
+
+    def apply_pagination_to_data_source(self, data_source: HttpAPIDataSource) -> HttpAPIDataSource:
+        if self.cursor:
+            cursor_params = {self.cursor_name: self.cursor}
+            if data_source.params is None:
+                data_source.params = cursor_params
+            else:
+                data_source.params |= cursor_params
+        return data_source
+
+    def get_next_pagination_config(
+        self, result: Any, pagination_info: Any | None
+    ) -> Optional["CursorBasedPaginationConfig"]:
+        if pagination_info is None:
+            return None
+        if isinstance(pagination_info, str):
+            return self.model_copy(update={"cursor": pagination_info})
+        else:
+            raise ValueError(f"Invalid next cursor value. Cursor value must be a string, got: {pagination_info}")
+
+    def get_pagination_info_filter(self) -> str:
+        return self.cursor_filter
+
+
+class PaginationType(str, Enum):
+    nothing = "nothing"
+    offset_limit = "offset_limit"
+    cursor = "cursor"
+    page = "page"
+
+
+class HttpPagination(BaseModel):
+    type: PaginationType = Field(
+        PaginationType.nothing,
+        description="The kind of pagination that corresponds to your API pagination system",
+    )
+    kwargs: dict = Field(
+        default_factory=dict,
+        title="Named arguments",
+        description="A JSON object with argument name as key and corresponding value as value",
+    )
+
+    def get_pagination_config(self) -> PaginationConfig:
+        auth_class = {
+            PaginationType.nothing: PaginationConfig,
+            PaginationType.offset_limit: OffsetLimitPaginationConfig,
+            PaginationType.cursor: CursorBasedPaginationConfig,
+            PaginationType.page: PageBasedPaginationConfig,
+        }[self.type]
+        pagination_config = auth_class(**self.kwargs)
+        return pagination_config
```

Comment:
Why not an union type ?

`HttpPaginationConfig: TypeAlias = PaginationConfig | OffsetLimitPaginationConfig ...`

Reply:
That may also lead to a better generated form, with arguments being more explicit for each type.